clang/lib/AST/ExprClassification.cpp

```diff
@@ -119,8 +119,6 @@ static Cl::Kinds ClassifyInternal(ASTContext &Ctx, const Expr *E) {
 
     // First come the expressions that are always lvalues, unconditionally.
   case Expr::ObjCIsaExprClass:
-    // C++ [expr.prim.general]p1: A string literal is an lvalue.
-  case Expr::StringLiteralClass:
     // @encode is equivalent to its string
   case Expr::ObjCEncodeExprClass:
```

Comment:
I think we also need to handle ObjCEncodeExprClass the same way.

Reply:
Also PredefinedExprClass in isTransparent mode (with -fms-extensions).

Reply:
Done.

For PredefinedExprClass, it doesn't repro on the original scenario, since we delay building those.